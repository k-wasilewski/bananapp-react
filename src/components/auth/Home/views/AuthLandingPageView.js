import React, {Component} from 'react';

class AuthLandingPageView extends Component {
    render() {
        return (
            <div>
                <header>
                    <input type='file' name='avatar' onChange={this.props.fileChangedHandler} />
                    <button type='button' onClick={this.props.submitLoading} > Upload </button>
                    { this.props.loadingComponent }
                    { this.props.errorMsg }
                    { this.props.$imagePreview }
                </header>
            </div>
        );
    }
}
export default AuthLandingPageView;