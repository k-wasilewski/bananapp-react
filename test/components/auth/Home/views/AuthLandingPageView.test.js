import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../../../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import AuthLandingPageView from "../../../../../src/components/auth/Home/views/AuthLandingPageView";
import {Provider} from "react-redux";

describe("AuthLandingPageView rendering specification", () => {
    it('AuthLandingPageView is rendered', () => {
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <AuthLandingPageView />
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});