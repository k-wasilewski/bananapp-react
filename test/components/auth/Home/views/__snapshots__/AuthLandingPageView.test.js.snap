// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLandingPageView rendering specification AuthLandingPageView is rendered 1`] = `
<div>
  <header>
    <input
      name="avatar"
      type="file"
    />
    <button
      type="button"
    >
       Upload 
    </button>
  </header>
</div>
`;
